File: RecyclerViewJava/app/src/main/java/com/devwilly/recyclerview_java/viewholders/FruitViewHolder.java
package com.devwilly.recyclerview_java.viewholders;

import com.devwilly.recyclerview_java.R;
import com.devwilly.recyclerview_java.wrapper.FruitWrapper;
import com.devwilly.recyclerview_java.wrapper.IEatingType;

import android.view.View;
import android.widget.TextView;


/**
 * Created by Willy on 15/01/2018.
 */

public class FruitViewHolder extends BaseViewHolder<FruitWrapper> {

    private TextView mFruitName;
    private TextView mFruitDescription;
    private TextView mFruitPrice;

    public FruitViewHolder(View itemView) {
        super(itemView);
        mFruitName = itemView.findViewById(R.id.fruit_name);
        mFruitDescription = itemView.findViewById(R.id.fruit_description);
        mFruitPrice = itemView.findViewById(R.id.fruit_price);
    }

    @Override
    public void bindView(FruitWrapper wrapper) {
        mFruitName.setText(wrapper.getName());
        mFruitDescription.setText(wrapper.getDescription());
        mFruitPrice.setText(wrapper.getPrice());
    }
}

File: RecyclerView-Kotlin/app/src/main/java/com/devwilly/recyclerviewkotlin/viewholders/FruitViewHolder.kt
package com.devwilly.recyclerviewkotlin.viewholders

import android.view.View
import android.widget.TextView
import com.devwilly.recyclerviewkotlin.R
import com.devwilly.recyclerviewkotlin.wrapper.FruitWrapper
import com.devwilly.recyclerviewkotlin.wrapper.IEatingWrapper

/**
 * Created by Willy on 15/01/2018.
 */
class FruitViewHolder(itemView: View) : BaseViewHolder<IEatingWrapper>(itemView) {

    private val mFruitName: TextView
    private val mFruitDescription: TextView
    private val mFruitPrice: TextView

    init {
        mFruitName = itemView.findViewById(R.id.fruit_name)
        mFruitDescription = itemView.findViewById(R.id.fruit_description)
        mFruitPrice = itemView.findViewById(R.id.fruit_price)

    }

    override fun bindView(t: IEatingWrapper) {
        mFruitName.text = t.getName()
        mFruitDescription.text = t.getDescription()
        mFruitPrice.text = t.getPrice()
    }
}